topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -961,25 +971,39 @@ private void updateStatusForException(ReconcilableTopic reconcilableTopic, Excep
             LOGGER.warnCr(reconcilableTopic.reconciliation(), "Updating status for unexpected exception", e);
             reason = e.getClass().getSimpleName();
         }
-        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), reason, message);
+        Condition condition = new ConditionBuilder()
+            .withType("Ready")
+            .withStatus("False")
+            .withReason(reason)
+            .withMessage(message)
+            .withLastTransitionTime(StatusUtils.iso8601Now())
+            .build();
+        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), condition);
     }
 
     private void updateStatusOk(ReconcilableTopic reconcilableTopic) {
-        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), null, null);
+        Condition condition = new ConditionBuilder()
+            .withType(isPaused(reconcilableTopic.kt()) ? "ReconciliationPaused" : "Ready")
+            .withStatus("True")
+            .withLastTransitionTime(StatusUtils.iso8601Now())
+            .build();
+        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), condition);
     }
 
     private void updateStatus(Reconciliation reconciliation,
                               KafkaTopic kt,
-                              String newReason,
-                              String newMessage) {
+                              Condition condition) {
         var oldStatus = kt.getStatus();
         Condition oldReadyCondition = oldStatus == null || oldStatus.getConditions() == null ? null : oldStatus.getConditions().stream().findFirst().orElse(null);
-        String newStatus = newReason != null || newMessage != null ? "False" : "True";
         if (oldStatus == null
                 || oldStatus.getObservedGeneration() != kt.getMetadata().getGeneration()
                 || oldStatus.getTopicName() == null
                 || oldReadyCondition == null
-                || isDifferentCondition(oldReadyCondition, newReason, newMessage, newStatus)) {
+                || isDifferentCondition(oldReadyCondition, condition)
+                || isPaused(kt)) {
```

Comment:
Won't this mean we do the update every time when a CR is paused? Isn't the `isDifferentCondition()` check enough to mean we will update the status when the annotation is paused/unpaused?

Reply:
I think you are right, but I need to update the type check in `isDifferentCondition()`.